includes/forms/class-llms-form-handler.php

```diff
@@ -331,6 +335,10 @@ public function submit( $posted_data, $location, $args = array() ) {
 		// Make sure the user id cannot be forced by user submission.
 		unset( $posted_data['user_id'] );
 
+		if ( ! empty( $args['validate_only'] ) ) {
+			return $this->validate_fields( $posted_data, $location, $fields, $action );
+		}
+
```

Comment:
I've explored a few different ways to enable separating validation from the form submission (so that we can validate all the form data without actually processing it) and this seems to be a pretty good way to do it. It doesn't require much new code to be written or modified. I do worry that adding it as an argument option buries it a little but it's clear enough (and I added the `llms_validate_user()` function to expose it less densely as well.

Reply:
I wonder if the `$action` above (used in the hook names) should be changed to adding `_validate` when validating only.
Like 
```php
		if ( ! empty( $args['validate_only'] ) ) {
			$action .= '_validate';
			return $this->validate_fields( $posted_data, $location, $fields, $action );
		}
```
?

Reply:
^

Reply:
What if then `$action` is mixed => `string|array( string $name, bool $validate_only )` and we pass this `validate_only` as additional parameter to the hooks?

Reply:
Because this hook, for example, is fired just before an user is created/updated, but now it can also be fired while only validating:
https://github.com/gocodebox/lifterlms/blob/9286df048b0063b499bff9a81cdc7271cc216d66/includes/forms/class-llms-form-handler.php#L483
And this seems odd to me, I'd expect that a user would be created right after but it won't.

Reply:
Okay... but still nothing has changed here. Previously this hook would have the same behavior, except that we didn't have a separate condition for *only validating*.

By this I mean if you as a developer *expect* that this hook is immediately followed by a user registration/update you're expecting something incorrectly because validation could still fail after this runs. 

Semantically sure the action name maybe doesn't make sense after this change but do we really need to provide every shred of internal context to every single hook we create?

In order to provide this context we need a SECOND action parameter to be passed to the function and in other contexts that parameter doesn't really make sense...

Adding a 5th parameter in order to pass this to a single action because the context *maybe* doesn't make sense seems like it's wholly unnecessary to me. I wa